pyperf/_psutil_memory.py

```diff
@@ -0,0 +1,44 @@
+try:
+    import os
+    import psutil
+    import threading
+    import time
+except ImportError:
+    raise ImportError('psutil is not installed')
+
+
+class PeakMemoryUsageThread(threading.Thread):
+    def __init__(self):
+        threading.Thread.__init__(self)
+        self.process = psutil.Process(os.getpid())
+        self.peak_usage = 0
+        self._done = threading.Event()
+        self.sleep = 0.001   # 1 ms
```

Comment:
It may use too much CPU time, a worker process runs at least 300 ms. Sampling every 10 ms should be enough.

Oh. pyperf/_memory.py also uses 1 ms. Maybe it should be increased as well :-)

What do you think?

Reply:
Agree, I've updated the sampling period as 10ms ;)